File: demo/python/trade_demo.py
import time
import requests
import hmac
from hashlib import sha256

APIURL = "https://open-api.bingx.com"
APIKEY = "Set your api key here !!"
SECRETKEY = "Set your secret key here!!"

def get_sign(api_secret, payload):
    signature = hmac.new(api_secret.encode("utf-8"), payload.encode("utf-8"), digestmod=sha256).hexdigest()
    print("sign=" + signature)
    return signature


def send_request(methed, path, urlpa, payload):
    url = "%s%s?%s&signature=%s" % (APIURL, path, urlpa, get_sign(SECRETKEY, urlpa))
    print(url)

    headers = {
        'X-BX-APIKEY': APIKEY,
    }

    response = requests.request(methed, url, headers=headers, data=payload)
    return response.text

def praseParam(paramsMap):
    sortedKeys = sorted(paramsMap)
    paramsStr = "&".join(["%s=%s" % (x, paramsMap[x]) for x in sortedKeys])
    return paramsStr

def cacel_order():
    payload = {}
    path = '/openApi/swap/v2/trade/order'
    methed = "DELETE"
    paramsMap = {
        "orderId": "1622560",
        "timestamp": int(time.time() * 1000),
        "symbol": "LINK-USDT",
    }
    paramsStr = praseParam(paramsMap)
    return send_request(methed, path, paramsStr, payload)

def get_balance():
    payload = {}
    path = '/openApi/swap/v2/user/balance'
    methed = "GET"
    paramsMap = {
        "timestamp": int(time.time() * 1000),
    }
    paramsStr = praseParam(paramsMap)
    return send_request(methed, path, paramsStr, payload)

def post_market_order():
    payload = {}
    path = '/openApi/swap/v2/trade/order'
    methed = "POST"
    paramsMap = {
        "side": "BUY",
        "positionSide": "LONG",
        "quantity": 5,
        "symbol": "LINK-USDT",
        "type": "MARKET",
        "timestamp": int(time.time() * 1000),
    }
    paramsStr = praseParam(paramsMap)
    return send_request(methed, path, paramsStr, payload)

if __name__ == '__main__':

    print("cacelOrder:", cacel_order())

    print("getBalance:", get_balance())

    print("postMarketOrder:", post_market_order())

File: demo/go/trade_demo.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/hex"
	"fmt"
	"io/ioutil"
	"net/http"
	"net/url"
	"time"
)

const (
	urlStr    = "https://open-api.bingx.com"
	apiKey    = "Set your api key here !!"
	secretKey = "Set your secret key here!!"
)

func computeHmac256(strMessage string, strSecret string) string {
	key := []byte(strSecret)
	h := hmac.New(sha256.New, key)
	h.Write([]byte(strMessage))

	return hex.EncodeToString(h.Sum(nil))
}

func send_request(requestUrl, method string) (responseData []byte, err error) {
	u, _ := url.Parse(requestUrl)
	req := http.Request{
		Method: method,
		URL:    u,
		Header: http.Header{},
	}
	req.Header.Set("X-BX-APIKEY", apiKey)
	cli := http.Client{
		Timeout: 3 * time.Second,
	}
	response, err := cli.Do(&req)
	if err != nil {

		return
	}
	defer response.Body.Close()

	responseData, err = ioutil.ReadAll(response.Body)
	return
}

func placeMarketOrder(symbol, side, quantity, positionSide string) {
	requestPath := "/openApi/swap/v2/trade/order"
	requestUrl := urlStr + requestPath
	timestamp := time.Now().UnixNano() / int64(time.Millisecond)
	mapParams := url.Values{}
	mapParams.Set("symbol", symbol)
	mapParams.Set("side", side)
	mapParams.Set("type", "MARKET")
	mapParams.Set("positionSide", positionSide)
	mapParams.Set("quantity", quantity)
	mapParams.Set("timestamp", fmt.Sprint(timestamp))
	strParams := mapParams.Encode()
	strParams += "&signature=" + computeHmac256(strParams, secretKey)
	requestUrl += "?"
	requestUrl += strParams
	responseData, err := send_request(requestUrl, http.MethodPost)
	fmt.Println("\trequest:", requestUrl)
	fmt.Println("\tresponse:", string(responseData), err)
}

func getBalance() {
	requestPath := "/openApi/swap/v2/user/balance"
	requestUrl := urlStr + requestPath
	timestamp := time.Now().UnixNano() / int64(time.Millisecond)
	mapParams := url.Values{}
	mapParams.Set("timestamp", fmt.Sprint(timestamp))
	strParams := mapParams.Encode()
	strParams += "&signature=" + computeHmac256(strParams, secretKey)
	requestUrl += "?"
	requestUrl += strParams
	responseData, err := send_request(requestUrl, http.MethodGet)
	fmt.Println("\trequest:", requestUrl)
	fmt.Println("\tresponse:", string(responseData), err)
}

func main() {
	fmt.Println("placeMarketOrder:")
	placeMarketOrder("LINK-USDT", "BUY", "5", "LONG")

	fmt.Println("getBalance:")
	getBalance()
}
